Automattic/wp-calypso
# Checkout: add dialog for info on upgrade credits

Related to Automattic/growth-foundations#143, Automattic/growth-foundations#141

## Proposed Changes

* This PR adds a (?) icon which opens a tooltip(popover) with information about upgrade credits. The term "Upgrade Credit" can be difficult to understand for new users and for users who directly land on the checkout page from in-product or email CTAs.

<img width="699" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/2446c0e3-104b-46e6-b5eb-bb26583a8cf3">

<img width="809" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/39d7ab7b-45c2-4ef8-bcd0-fe49316fc2be">




* We do not have a design for this yet, but hopefully, this PR can serve as a foundation, and the design can be polished.

## Testing Instructions



* On a site with a paid plan, go to `/plans/<site slug>` and click on any of the "Upgrade" buttons.
* On the checkout page, confirm that you can see the help icon.
* Confirm that clicking the icon opens up the modal.

## Pre-merge Checklist
